core/src/jsMain/kotlin/external/JsIterator.kt

```diff
@@ -0,0 +1,25 @@
+package com.juul.kable.external
+
+internal external interface JsIterator<T> {
+    fun next(): JsIteratorResult<T>
+}
+
+internal external interface JsIteratorResult<T> {
+    val done: Boolean
+    val value: T?
+}
+
+internal fun <T> JsIterator<T>.iterable(): Iterable<T> {
```

Comment:
Sucks that we need to manually define this.

Reply:
My hope is the new IR compiler (that JetBrains is working on) will provide this for us. We shall see though.